src/dataflow/operators/ros/to_ros_operator.rs

```diff
@@ -0,0 +1,76 @@
+use crate::dataflow::{context::SinkContext, operator::Sink, Data};
+use serde::Deserialize;
+use std::sync::Arc;
+
+/// Takes an input ERDOS stream and publishes to a ROS topic using the provided message conversion
+/// function.
+///
+/// Conversion function should convert a Rust data type and return a ROS type which implements
+/// the [`rosrust::Message`] trait.
+///
+/// # Example
+/// The following example shows how to use a [`ToRosOperator`] with a conversion function which takes
+/// a Rust [`i32`] and converts it to a ROS message with [`rosrust_msg::std_msgs::Int32`] data.
+///
+/// Assume that `source_stream` is an ERDOS stream sending the correct messages.
+///
+/// ```
+/// fn erdos_int_to_ros_int(input: &i32) -> rosrust_msg::std_msgs::Int32 {
+///     rosrust_msg::std_msgs::Int32 { data: input.data }
+/// }
+///
+/// let ros_sink_config = OperatorConfig::new().name("ToRosInt32");
+/// erdos::connect_sink(
+///     move || -> ToRosOperator<i32, rosrust_msg::std_msgs::Int32> {
+///         ToRosOperator::new("int_topic", erdos_int_to_ros_int)
+///     },
+///     || {},
+///     ros_sink_config,
+///     &source_stream,
+/// );
+/// ```
+
+pub struct ToRosOperator<T, U: rosrust::Message>
+where
+    T: Data + for<'a> Deserialize<'a>,
+{
+    publisher: rosrust::Publisher<U>,
+    to_ros_msg: Arc<dyn Fn(&T) -> U + Send + Sync>,
+}
+
+impl<T, U: rosrust::Message> ToRosOperator<T, U>
```

Comment:
Nit: `ToROSOperator`?

Reply:
Let's go with `ToRosOperator`. Looks like it fits better with typical camel case usage:
https://stackoverflow.com/questions/15526107/acronyms-in-camelcase